manta-pay/src/key.rs

```diff
@@ -185,9 +195,9 @@ where
     where
         R: RngCore + ?Sized,
     {
-        let mut seed = [0; Seed::SIZE];
-        rng.fill_bytes(&mut seed);
-        Self::build(seed)
+        let sample_mnemonic = Mnemonic::sample(&mut OsRng);
+        let seed = sample_mnemonic.to_seed("");
+        Self::build(*seed.as_bytes(), sample_mnemonic)
```

Comment:
```suggestion
        Self::new(Mnemonic::sample(rng), "")
```

Reply:
I think we need to use OsRng instead of the rng passed to the sample trait, because Mnemonic::sample() requires CryptoRng and RngCore traits but rng only implements RngCore